sprockets/mixins/metrics/statsd.py

```diff
@@ -128,10 +160,17 @@ def send(self, path, value, metric_type):
         """
         msg = '{0}:{1}|{2}'.format(
             self._build_path(path, metric_type), value, metric_type)
+
+        encoding = 'utf-8'
+        #encoding = 'ascii'
+
         try:
-            LOGGER.debug('Sending %s to %s:%s', msg.encode('ascii'),
+            LOGGER.debug('Sending %s to %s:%s', msg.encode(encoding),
                          self._host, self._port)
-            self._sock.sendto(msg.encode('ascii'), (self._host, self._port))
+            if self._tcp:
+                return self._sock.write(msg.encode(encoding))
+            else:
+                self._sock.sendto(msg.encode(encoding), (self._host, self._port))
         except socket.error:
```

Comment:
Why are you concerned about a socket.error in your debug log?

Reply:
that was the existing except in sprokets, im going to modify to match https://github.com/gmr/rejected/blob/3.19/rejected/statsd.py#L110 so it logs something meaningful :
 
        except (OSError, socket.error) as error:  # pragma: nocover
            LOGGER.exception('Error sending statsd metric: %s', error)

Reply:
my point is that your debug logging doesn't belong in the try/except block.